itc4j-jackson/src/main/java/itc4j/NonLeafEventSerializer.java

```diff
@@ -0,0 +1,31 @@
+package itc4j;
+
+import com.fasterxml.jackson.core.*;
+import com.fasterxml.jackson.databind.jsontype.TypeSerializer;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+
+import java.io.IOException;
+
+/**
+ * Serialize {@link itc4j.NonLeafEventSerializer}
+ *
+ * @author Ian Eure <ian.eure@gmail.com>
+ */
+public class NonLeafEventSerializer extends StdSerializer<NonLeafEvent> {
+  public NonLeafEventSerializer() {
+    super(NonLeafEvent.class);
+  }
+
+  @Override
+  public void serialize(final NonLeafEvent id,
+                        final JsonGenerator jgen,
+                        final SerializerProvider provider)
+  throws IOException {
+    jgen.writeStartArray();
+    jgen.writeNumber(id.getValue());
+    jgen.writeObject(id.getLeft());
+    jgen.writeObject(id.getRight());
+    jgen.writeEndArray();
+  }
```

Comment:
It's more compact, reduces repetition of keys, and matches what `toString` does.

`Stamp.toString()`: `(((0, (1, 0)), 0), (1, (0, 0, (2, 1, 0)), 0))`
JSON representation: `[[[0,[1,0]],0],[1,[0,0,[2,1,0]],0]]`

Using an object would produce a much more verbose representation, something like:
(excuse errors, this is just a by-hand SWAG)
```json
{"id": {"left": {"left": 0,
                 "right": {"left": 1, "right": 0}},
        "right": 0},
 "event": …
}
```

I don't think this is more readable for humans, because the density of actual data to labels is very low, the payload size is significantly larger, and there are no benefits when writing the ser/deser code.